moxy-compiler/src/main/java/moxy/compiler/viewstateprovider/InjectViewStateProcessor.kt

```diff
@@ -11,25 +11,44 @@ import moxy.compiler.Util
 import moxy.compiler.Util.fillGenerics
 import moxy.compiler.asTypeElement
 import moxy.compiler.getFullClassName
+import javax.lang.model.element.Modifier
 import javax.lang.model.element.TypeElement
 import javax.lang.model.element.TypeParameterElement
 import javax.lang.model.type.MirroredTypeException
 import javax.lang.model.type.TypeKind
 import javax.lang.model.type.TypeMirror
 
-class InjectViewStateProcessor : ElementProcessor<TypeElement, PresenterInfo> {
+class InjectViewStateProcessor : ElementProcessor<TypeElement, PresenterInfo?> {
 
+    /**
+     * A ViewState implementation is generated for each
+     * View interface in this set.
+     *
+     * This set is filled during the processing of presenters
+     * annotated with @InjectViewState.
+     */
     val usedViews = mutableSetOf<TypeElement>()
 
-    override fun process(element: TypeElement): PresenterInfo {
-        return PresenterInfo(element, getViewStateClassName(element)!!)
+    /**
+     * Returns null if this Presenter should not be used for ViewState and
+     * ViewStateProvider generation.
+     */
+    override fun process(element: TypeElement): PresenterInfo? {
+        return getViewStateClassName(element)?.let { PresenterInfo(element, it) }
     }
 
+    /**
+     * Returns null if this Presenter should not be used for ViewState generation.
+     */
     private fun getViewStateClassName(typeElement: TypeElement): String? {
         val viewState = getViewStateClassFromAnnotationParams(typeElement)
         if (viewState != null) return viewState
 
-        val view = getViewClassFromPresenterTypeElement(typeElement)
+        if (typeElement.isAbstractClass) {
+            return null
```

Comment:
We can check for abstract class earlier in `process()`, and make `getViewStateClassName()` return `String` instead of `String?`

Reply:
Good point, fixed.